servicetalk-concurrent-api/src/testFixtures/java/io/servicetalk/concurrent/api/TerminalSignalConsumerMock.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright © 2020 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.concurrent.api;
+
+import org.mockito.Mockito;
+
+import static org.mockito.AdditionalAnswers.delegatesTo;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+/**
+ * Mocked {@link TerminalSignalConsumer} that helps to verify that only a single terminal method was invoked.
+ */
+public final class TerminalSignalConsumerMock implements TerminalSignalConsumer {
+
+    private Runnable runnable;
+    private TerminalSignalConsumer signalConsumer;
+
+    /**
+     * Creates a new instance.
+     */
+    public TerminalSignalConsumerMock() {
+        runnable = Mockito.mock(Runnable.class);
```

Comment:
What is the use of the `Runnable` here?

Reply:
@NiteshKant I think it is just used in the line below. @idelpivnitskiy consider to adjust scope to constructor.

Reply:
Trying to test that `RunnableTerminalSignalConsumer` delegates to this `Runnable` correctly and invokes it only once.